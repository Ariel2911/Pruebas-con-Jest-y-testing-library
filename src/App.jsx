import PropTypes from 'prop-types';

function App({ title, subtitle, name }) {
  return (
    <>
      <h1>{title}</h1>
      <p>{subtitle}</p>
      <p>{name}</p>
    </>
  );
}

export default App;

App.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  name: PropTypes.string,
};

App.defaultProps = {
  // title: 'No hay título',
  subtitle: 'No hay subtítulo',
  name: 'Ariel',
};
