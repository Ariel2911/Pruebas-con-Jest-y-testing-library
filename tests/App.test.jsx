import { render } from '@testing-library/react';
import App from '../src/App';

describe('Pueba en <App />', () => {
  test('debe de hacer mach con el snapshop', () => {
    const title = 'Hola!';
    const { container } = render(<App title={title} />);

    expect(container).toMatchSnapshot();
  });
});
