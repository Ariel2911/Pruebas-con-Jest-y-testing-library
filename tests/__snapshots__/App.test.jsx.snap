// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pueba en <App /> debe de hacer mach con el snapshop 1`] = `
<div>
  <h1>
    Hola!
  </h1>
  <p>
    No hay subt√≠tulo
  </p>
  <p>
    Ariel
  </p>
</div>
`;
